cabal-install/Distribution/Client/ProjectConfig/Types.hs

```diff
@@ -186,6 +189,18 @@ data ProjectConfigShared
   deriving (Eq, Show, Generic)
 
 
+-- | Specifies the provenance of project configuration, whether defaults were
+-- used or if the configuration was read from an explicit file path.
+data ProjectConfigProvenance
+
+     -- | The project configuration was implied.
```

Comment:
Implied by what? If I read this, I would have no idea where to look to find out more. Crossref please.